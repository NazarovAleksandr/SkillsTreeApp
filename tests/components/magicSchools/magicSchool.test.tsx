import * as React from 'react';
import { shallow } from 'enzyme';
import MagicSchool from '../../../src/components/magicSchools/magicSchool';

describe('MagicSchool', () => {
    test('Renders passive display', () => {
        const school = { name: 'testSchool', id: '1' };
        const component = shallow(<MagicSchool school={school} />);

        expect(component).toMatchSnapshot('passiveDisplay');
    });
    test('Renders active school', () => {
        const school = { name: 'testSchool', id: '1' };
        const component = shallow(<MagicSchool school={school} isActive />);

        expect(component).toMatchSnapshot('activeDisplay');
    });
    test('Renders is edition state', () => {
        const school = { name: 'testSchool', id: '1' };
        const component = shallow(<MagicSchool school={school} inEdition />);

        expect(component).toMatchSnapshot('inEditionState');
    });
    test('Calls select callback on click', () => {
        const school = { name: 'testSchool', id: '1' };
        const fn = jest.fn();

        const component = shallow(<MagicSchool school={school} inEdition onSelect={fn} />);
        component.find('.school').simulate('click');

        expect(fn).toBeCalled();
    });
    test('Calls callback on input bluring', () => {
        const school = { name: 'testSchool', id: '1' };
        const fn = jest.fn();

        const component = shallow(<MagicSchool school={school} inEdition onEditionEnd={fn} />);
        component.find('.input').simulate('blur');

        expect(fn).toBeCalled();
    });
});
