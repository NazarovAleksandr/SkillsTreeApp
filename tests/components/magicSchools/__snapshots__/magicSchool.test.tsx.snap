// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicSchool Renders active school: activeDisplay 1`] = `
<div
  className="school active"
  role="button"
  tabIndex={0}
>
  <span
    className="title"
    title="testSchool"
  >
    testSchool
  </span>
</div>
`;

exports[`MagicSchool Renders is edition state: inEditionState 1`] = `
<div
  className="school"
  role="button"
  tabIndex={0}
>
  <div
    className="active-container"
  >
    <input
      autoFocus={true}
      className="input"
      defaultValue="testSchool"
    />
  </div>
</div>
`;

exports[`MagicSchool Renders passive display: passiveDisplay 1`] = `
<div
  className="school"
  role="button"
  tabIndex={0}
>
  <span
    className="title"
    title="testSchool"
  >
    testSchool
  </span>
</div>
`;
